substratest/assets.py

```diff
@@ -50,6 +50,50 @@ def get(self):
         return self._asset
 
 
+class ComputePlanFuture(Future):
```

Comment:
What do you think about creating an abstract base class `BaseFuture` with two abstract methods `get()` and `wait()`.

Here, it seems that `ComputePlanFuture` does not really need to inherit from `Future`.

Reply:
Definitely!